content/blog/2017-06-21-prometheus-20-alpha3-new-rule-format.md

```diff
@@ -0,0 +1,71 @@
+---
+ title: Prometheus 2.0 Alpha.3 with New Rule Format
+ created_at: 2017-06-21
+ kind: article
+ author_name: Goutham Veeramachaneni
+---
+
+Today we release the third alpha version of Prometheus 2.0. Aside from a variety of bug fixes in the new storage layer, it contains a few planned breaking changes.
+ 
+First, we moved to a new flag library, which uses the more common double-dash `--` prefix for flags instead of the single dash Prometheus used so far. Deployments have to be adapted accordingly.
+Additionally, some flags were removed with this alpha. The full list since Prometheus 1 is:
+ 
+`web.telemetry-path`<br/>
+All `storage.remote.*` flags<br/>
+All `storage.local.*` flags<br/>
+`query.staleness-delta`<br/>
+`alertmanager.url`<br/>
+ 
+Alerting and recording rules are one of the critical features of Prometheus. But they also come with a few design issues and missing features, namely:
+ 
+* All rules ran with the same interval. We could have some heavy rules that are better off being run at a 10-minute interval and some rules that could be run at 15-second intervals.
+
+* All rules were evaluated concurrently, which is actually Prometheus’ oldest [open bug](https://github.com/prometheus/prometheus/blob/master/rules/manager.go#L267). This has a couple of issues, the obvious one being that the load spikes every eval interval if you have a lot of rules. The other being that rules that depend on each other might be fed outdated data. For example:
+
+```
+instance:network_bytes:rate1m = sum by(instance) (rate(network_bytes_total[1m]))
+
+ALERT HighNetworkTraffic
+  IF instance:network_bytes:rate1m > 10e6
+  FOR 5m
+```
+
+
+Here we are alerting over `instance:network_bytes:rate1m`, but `instance:network_bytes:rate1m` is itself being generated by another rule. We can get proper results only if the alert `HighNetworkTraffic` is run after the current value for `instance:network_bytes:rate1m` gets recorded.
```

Comment:
I'd say expected rather than proper. The result is proper, it just may be delayed by a cycle.